### ISSUE TITLE
network-firewall-policy not able to use target_secure_tags

### ISSUE THREAD

**Issue by Samir-Cit** [terraform-google-modules/terraform-google-network#505]
### TL;DR

Not able to set **target_secure_tags** while creating a **network-firewall-policy**.
The module always set the variable **target_service_accounts** as an _empty list []_ and the **target_secure_tags** can only be set if the variable **target_service_accounts** is **_null_**.

### Expected behavior

Create the Firewall Policy Rule with the **target_secure_tags**.

To do so, the variable **target_service_accounts** value need to be set as null.
As the following code:

REPLACE THIS:
`target_service_accounts = optional(list(string), [])`
FOR THIS:
`target_service_accounts = optional(list(string))`

### Observed behavior

The module creates the Firewall Policy Rule without the **target_secure_tags**.
The following block is the plan output.

```hcl
# module.firewall_tags.module.firewall_rules.google_compute_network_firewall_policy_rule.rules["1000"] will be created
  + resource "google_compute_network_firewall_policy_rule" "rules" {
      + action                  = "allow"
      + direction               = "INGRESS"
      + enable_logging          = true
      + firewall_policy         = "some-policy"
      + id                      = (known after apply)
      + kind                    = (known after apply)
      + priority                = 1000
      + project                 = "some-project"
      + rule_name               = "some-name"
      + rule_tuple_count        = (known after apply)
      + target_service_accounts = []

      + match {
          + dest_address_groups       = []
          + dest_fqdns                = []
          + dest_ip_ranges            = []
          + dest_region_codes         = []
          + dest_threat_intelligences = []
          + src_address_groups        = []
          + src_fqdns                 = []
          + src_ip_ranges             = [
              + "35.235.240.0/20",
            ]
          + src_region_codes          = []
          + src_threat_intelligences  = []

          + layer4_configs {
              + ip_protocol = "tcp"
              + ports       = [
                  + "22",
                ]
            }
        }
    }
```

### Terraform Configuration

```hcl
data "google_netblock_ip_ranges" "iap_forwarders" {
  range_type = "iap-forwarders"
}

module "firewall_rules" {
  source       = "terraform-google-modules/network/google//modules/network-firewall-policy"
  version      = "~> 7.2"

  project_id   = "some-project"
  policy_name  = "some-policy"
  target_vpcs  = ["some-network"]

  rules = [
    {
      rule_name      = "some-name"
      action         = "allow"
      direction      = "INGRESS"
      priority       = "1000"
      enable_logging = true
      target_secure_tags = ["tagValues/${google_tags_tag_value.firewall_tag_value.name}"]
      match = {
        src_ip_ranges = data.google_netblock_ip_ranges.iap_forwarders.cidr_blocks_ipv4
        layer4_configs = [
          {
            ip_protocol = "tcp"
            ports       = ["22"]
          },
        ]
      }
    }
  ]

  depends_on = [
    google_tags_tag_value.firewall_tag_value
  ]
}

resource "google_tags_tag_key" "firewall_tag_key" {
  short_name = "tag-key"
  parent     = "projects/some-project"
  purpose    = "GCE_FIREWALL"

  purpose_data = {
    network = "some-project/some-network"
  }
}

resource "google_tags_tag_value" "firewall_tag_value" {
  short_name  = "tag-value"
  parent      = "tagKeys/${google_tags_tag_key.firewall_tag_key.name}"
}
```

### Terraform Version

```sh
Terraform v1.3.0
on linux_amd64
+ provider registry.terraform.io/hashicorp/google v4.83.0
+ provider registry.terraform.io/hashicorp/google-beta v4.84.0
```
